models/payment_sync.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+import logging
+import requests
+import xml.etree.ElementTree as ElementTree
+from datetime import datetime
+from odoo import api, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentSync(models.Model):
+    _name = "payment.sync"
+
+    @api.model
+    def sync_payments(self):
+        """ Syncs Odoo payments to CiviCRM
+         :return:
+        """
+        _logger.debug("Payment Sync Started")
+        payments = self._get_payments()
+        if payments:
+            self._process_payments(payments)
+        else:
+            _logger.debug("No payments were found")
+
+    def _process_payments(self, payments):
+        """ Processing payments sync
+         :param payments: list of payments
+         :return: list of failed payments
+        """
+        for payment in payments:
+            if self._sync(payment):
+                self._send_report(payment)
+
+    def _sync(self, payment):
+        """ Syncs single record of Payment to CiviCRM
+         :param payment: account_payment model
+         :return: bool True on success, False on failure
+        """
+        data = self._fill_sync_data(payment)
```

Comment:
please move this line to be right before : 

```php
xml_doc = self._create_xml_with_data(data)
```

since variables should be declared near to where they first used.